packages/expo/src/Expo.fx.tsx

```diff
@@ -28,7 +28,7 @@ if (NativeModulesProxy.ExpoUpdates?.isMissingRuntimeVersion) {
   }
 }
 
-if (Constants.executionEnvironment === ExecutionEnvironment.StoreClient) {
+if (isRunningInExpoGo()) {
```

Comment:
nice 🤔 we should maybe remove `Constants.executionEnvironment` entirely, and recommend people use this API instead

Reply:
`expo-constants` is currently a big mess:

- a lots of deprecated fields
- some have missing description in the docs
- some are no longer used
- some shouldn't even be public

I think long term we can even remove that package entirely – we mostly need the manifest, but this can be exposed by `expo-updates` or `expo-dev-client`